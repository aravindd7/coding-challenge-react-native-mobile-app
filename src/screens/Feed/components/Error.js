import React from "react";
import { Text, View, TouchableOpacity } from "react-native";

export default ({ error, feedUpdate }) => (
  <View>
    <Text>Ha ocurrido un error inesperado</Text>
    <Text>{error}</Text>
    <TouchableOpacity onPress={feedUpdate}>
      <Text>Try again</Text>
    </TouchableOpacity>
  </View>
);
