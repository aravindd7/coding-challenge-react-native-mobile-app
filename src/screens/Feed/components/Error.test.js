import React from "react";

import ShallowRenderer from "react-test-renderer/shallow"; // ES6

import Error from "./Error";

const props = {
    error: 'error message',
    feedUpdate: () => {},
};


it("match on Shallow mode", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Error {...props} />);
  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});



