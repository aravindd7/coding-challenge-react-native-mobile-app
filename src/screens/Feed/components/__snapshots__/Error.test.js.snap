// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match on Shallow mode 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Ha ocurrido un error inesperado
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    error message
  </Text>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Try again
    </Text>
  </TouchableOpacity>
</View>
`;
